rfcs/0192-min-capacity-ensures-workers-do-not-get-unnecessarily-killed.md

```diff
@@ -0,0 +1,199 @@
+
+# Details
+
+## Current Behavior Analysis
+
+**Affected Worker Pools:**
+- Direct `minCapacity: 1`: [`infra/build-decision`](https://github.com/mozilla-releng/fxci-config/blob/43c18aab0826244e369b16a964637b6c411c7760/worker-pools.yml#L220), [`code-review/bot-gcp`](https://github.com/mozilla-releng/fxci-config/blob/43c18aab0826244e369b16a964637b6c411c7760/worker-pools.yml#L3320)
+- Keyed `minCapacity: 1`: [`gecko-1/decision-gcp`, `gecko-3/decision-gcp`](https://github.com/mozilla-releng/fxci-config/blob/43c18aab0826244e369b16a964637b6c411c7760/worker-pools.yml#L976), and [pools matching `(app-services|glean|mozillavpn|mobile|mozilla|translations)-1`](https://github.com/mozilla-releng/fxci-config/blob/43c18aab0826244e369b16a964637b6c411c7760/worker-pools.yml#L1088)
+
+**Current Implementation Issues:**
+- Worker-manager enforces `minCapacity` by spawning new workers when capacity drops below threshold
+- Generic-worker shuts down after `idleTimeoutSecs` regardless of `minCapacity` requirements
+- Gap exists between worker shutdown and replacement detection/provisioning
+
+## Proposed Solution
+
+### Core Concept: `minCapacity` Workers
+
+Workers fulfilling `minCapacity >= 1` requirements should receive significantly longer idle timeouts to preserve caches.
+
+### Implementation Approach
+
+#### 1. Worker Identification and Tagging
+
+**Worker Provisioning Logic:**
+- Worker-manager determines worker idle timeout configuration at spawn time based on current pool capacity
+- **No Special Replacement Logic**: Failed workers are handled through normal provisioning cycles, not immediate replacement
+
+#### 2. Lifecycle-Based Worker Configuration
+
+Workers are identified with a boolean flag at launch time indicating whether they fulfill minCapacity requirements. This approach works within TaskCluster's architectural constraints where worker-manager cannot communicate configuration changes to running workers.
+
+**Launch-Time Configuration:**
+Worker-manager determines worker type at spawn time based on current pool capacity needs. Workers fulfilling `minCapacity` requirements are marked with a boolean flag and receive indefinite idle timeout (0 seconds). Additional workers beyond `minCapacity` receive standard idle timeouts (default 600 seconds).
+
+**Immutable Worker Behavior:**
+- Workers receive their complete configuration at startup
+- No runtime configuration changes
+- Worker idle timeout never changes during worker lifetime
+- `minCapacity` requirements fulfilled through worker replacement, not reconfiguration
+
+#### 3. Pool Configuration
+
+**New Configuration Options:**
+Pools will have a new boolean flag `enableMinCapacityWorkers` that enables minCapacity worker behavior. When enabled, workers fulfilling minCapacity requirements will run indefinitely (idle timeout set to 0), providing maximum cache preservation.
```

Comment:
I wonder if we really want an extra flag for this. For me `minCapacity` is already there. 
Unless someone wanted to use it the way they were currently working, I think we can just not introduce a new flag. 
just  thinking out loud

Reply:
I was thinking of a new flag in order to roll out the changes. But maybe there's another way I didn't think of. I'm option to other options 🙂